### ISSUE TITLE
Kernel Crash on executing cell block having torch optimizer

### ISSUE THREAD

**Issue by adarshadan** [microsoft/vscode-jupyter#17396]
## Environment data

-   VS Code version: 1.120.0
-   Jupyter Extension version (available under the Extensions sidebar): 2025.9.1
-   Python Extension version (available under the Extensions sidebar): 2026.4.0
-   OS (Windows | Mac | Linux distro) and version: Windows 11 (Windows_NT x64 10.0.26200)
-   Python and/or Anaconda version: 3.12.2
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
-   Jupyter server running: N/A

## Expected behaviour

Cell block should get executed without kernel crashing

## Actual behaviour

Kernel died while executing cell block

## Steps to reproduce:

1. Install Libraries - 
%pip install pandas==2.2.2 numpy==1.26.4 matplotlib==3.8.0 scikit-learn==1.5.0 ucimlrepo==0.0.7
%pip install torch==2.8.0+cpu torchvision==0.23.0+cpu torchaudio==2.8.0+cpu \
    --index-url https://download.pytorch.org/whl/cpu
2. Define and instantiate model -
import torch.nn as nn
class ClassificationNet(nn.Module):
    def __init__(self, input_units=30, hidden_units=64, output_units=2):
        super(ClassificationNet, self).__init__()
        self.fc1 = nn.Linear(input_units, hidden_units)
        self.fc2 = nn.Linear(hidden_units, output_units)

    def forward(self, x):
        x = torch.relu(self.fc1(x))
        x = torch.softmax(self.fc2(x), dim=1)
        return x
model = ClassificationNet(input_units=30, hidden_units=64, output_units=2)

3. Execute below cell block to define loss and optimizer-
from torch import optim

criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=0.001)

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
Visual Studio Code (1.120.0, undefined, desktop)
Jupyter Extension Version: 2025.9.1.
Python Extension Version: 2026.4.0.
Python Environment Extension Version: 1.30.0.
Pylance Extension Version: 2026.2.1.
Platform: win32 (x64).
Home = c:\Users\adars
Temp Storage folder ~\AppData\Roaming\Code\User\globalStorage\ms-toolsai.jupyter\version-2025.9.1
Workspace folder ~\Desktop\Coursera AI Course\Labs\Introduction to Pytorch
18:21:33.883 [info] Starting Kernel (Python Path: ~\AppData\Local\Programs\Python\Python312\python.exe, Unknown, 3.12.2) for '~\Desktop\Coursera AI Course\Labs\Introduction to Pytorch\Deep Neural Network for Wine Classification-v1.ipynb' (disableUI=true)
18:21:37.425 [warn] Failed to get activated env vars for ~\AppData\Local\Programs\Python\Python312\python.exe in 636ms
18:21:37.675 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe -c "import site;print("USER_BASE_VALUE");print(site.USER_SITE);print("USER_BASE_VALUE");"
18:21:38.221 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe ~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\pythonFiles\vscode_datascience_helpers\kernel_interrupt_daemon.py --ppid 24560
    > cwd: ~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\pythonFiles\vscode_datascience_helpers
18:21:38.477 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
18:21:38.832 [warn] Failed to get activated env vars for ~\AppData\Local\Programs\Python\Python312\python.exe in 2016ms
18:21:39.118 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe -m pip list
18:21:39.474 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe -m ipykernel_launcher --f=~\AppData\Roaming\jupyter\runtime\kernel-v39bfb6c73aeb9e77b2b460ca18b9556902b251925.json
    > cwd: ~\Desktop\Coursera AI Course\Labs\Introduction to Pytorch
18:21:41.800 [info] Kernel successfully started
18:21:42.237 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe ~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\pythonFiles\printJupyterDataDir.py
18:22:38.547 [error] Disposing session as kernel process died ExitCode: 3221225477, Reason: 
18:34:42.888 [error] Failed to write data to the kernel channel shell [
  <Buffer 3c 49 44 53 7c 4d 53 47 3e>,
  <Buffer 39 38 62 38 36 34 33 31 64 39 35 36 66 30 33 32 36 39 62 63 64 30 33 38 39 30 30 35 35 32 61 33 32 37 39 34 32 36 65 36 66 33 31 35 36 61 65 66 35 32 ... 14 more bytes>,
  <Buffer 7b 22 64 61 74 65 22 3a 22 32 30 32 36 2d 30 35 2d 31 34 54 31 33 3a 30 34 3a 34 32 2e 38 38 37 5a 22 2c 22 6d 73 67 5f 69 64 22 3a 22 32 35 33 37 65 ... 177 more bytes>,
  <Buffer 7b 7d>,
  <Buffer 7b 7d>,
  <Buffer 7b 22 73 69 6c 65 6e 74 22 3a 66 61 6c 73 65 2c 22 73 74 6f 72 65 5f 68 69 73 74 6f 72 79 22 3a 66 61 6c 73 65 2c 22 75 73 65 72 5f 65 78 70 72 65 73 ... 12617 more bytes>
] [Error: Socket is closed
	at a.postToSocket (~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\dist\extension.node.js:309:7983)
	at ~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\dist\extension.node.js:309:7727] {
  errno: 9,
  code: 'EBADF'
}
18:34:42.889 [error] Failed to write data to the kernel channel shell [
  <Buffer 3c 49 44 53 7c 4d 53 47 3e>,
  <Buffer 34 64 65 31 65 63 66 37 37 31 36 35 66 62 30 35 64 38 61 64 62 38 31 37 39 30 65 38 35 62 65 35 34 32 37 32 35 31 30 65 63 38 63 66 38 31 34 63 62 35 ... 14 more bytes>,
  <Buffer 7b 22 64 61 74 65 22 3a 22 32 30 32 36 2d 30 35 2d 31 34 54 31 33 3a 30 34 3a 34 32 2e 38 38 38 5a 22 2c 22 6d 73 67 5f 69 64 22 3a 22 36 61 36 32 36 ... 177 more bytes>,
  <Buffer 7b 7d>,
  <Buffer 7b 7d>,
  <Buffer 7b 22 73 69 6c 65 6e 74 22 3a 66 61 6c 73 65 2c 22 73 74 6f 72 65 5f 68 69 73 74 6f 72 79 22 3a 66 61 6c 73 65 2c 22 75 73 65 72 5f 65 78 70 72 65 73 ... 12617 more bytes>
] [Error: Socket is closed
	at a.postToSocket (~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\dist\extension.node.js:309:7983)
	at ~\.vscode\extensions\ms-toolsai.jupyter-2025.9.1-win32-x64\dist\extension.node.js:309:7727] {
  errno: 9,
  code: 'EBADF'
}

<img width="1555" height="269" alt="Image" src="https://github.com/user-attachments/assets/c2af3fa3-1ef0-4677-9805-c04a8cc1587b" />

```

</p>
</details>

**Comment by DonJayamanne** [comment]
Thank you for filing this issue and sorry you are running into this.
We've found that such issues are caused by the packages itself, e.g. its possible you have a dated package, or its not compatible with another version of some other packge installed.
Closing this as Jupyter extension doesn't really manage the packages, it uses Python runtime to executed the code, ad the fact that it hangs is just a symptom of something else thats wrong with the environment.
